### PR TITLE
add cpp documentation

### PR DESCRIPTION
**Pull request by starlitxiling**
This is a cpp document written according to doxygen.

### CODE REVIEW

File: src/pages/index.js [hunk 1]
```diff
@@ -324,7 +324,7 @@ function HomepageHeader() {
           <div style={{ display: "flex", flexDirection: "row", justifyContent: "space-evenly", flexWrap: "wrap" }}>
             <div className={styles.buttons}>
               <Link
-                className="button button--secondary button--md margin-bottom--md"
+                className="button button--secondary button--md ma rgin-bottom--md"
```
Comment (haixuanTao) [inline src/pages/index.js:327]: what is this change about?

**Comment by haixuanTao** [comment]
The link is not loading when I build the website.

Can you add information on how to build the static files?

image

File: src/pages/index.js (hunk 1 above)
Comment (starlitxiling) [reply src/pages/index.js:327]: oh, maybe it change when i build,i will check this

**Comment by starlitxiling** [comment]
when you build at local machine,you should change the link to "http://localhost:3000/cpp/cpp-api.html"

**Comment by haixuanTao** [comment]
Ah ok, could it be possible to have the build script for the documentation?

File: src/pages/index.js [hunk 2]
```diff
@@ -337,7 +337,7 @@ function HomepageHeader() {
               >
                 <Translate id="homepage.python-api.button.text">🐍 Python API</Translate>
               </Link>
-            </div>
+            </div>
```
Comment (haixuanTao) [inline src/pages/index.js:340]: Can we remove this trailing white space?

**Comment by starlitxiling** [comment]
Ok, I delete search bar

**Comment by haixuanTao** [comment]
Could you please add the script to use to generate the documentation?

I can't merge this PR otherwise

**Comment by starlitxiling** [comment]
Do you mean the script of the cpp document? I use the Doxygen template, but the main content of the document is written in html, not generated from the Markdown document. If you want to modify the document, you only need to modify cpp-api.html. Of course, I provide a generated script https://github.com/dora-rs/dora-rs.github.io/pull/20/commits/4b6b79252b1a251abfb47ae2183e100486ad7db9#diff-f1cf1fbc3fe2fc86148347689881cb23764d3981c211d3bbd6573f9d5869622d, which can back up the main content of the cpp document and regenerate the document. You can use it to add some js code and regenerate the document.

**Comment by haixuanTao** [comment]
I don't understand is this documentation not generated from code?

Did you write it yourself?

**Comment by starlitxiling** [comment]
Yes, Doxygen can only generate documentation from C++ code, but dora's C++ interface is declared in rust, and files like bindings.cc are generated only when used, so I can only use this template to write the document myself.

**Comment by haixuanTao** [comment]
So, I opened a PR: https://github.com/dora-rs/dora/pull/611 to be able to auto generate C++ documentation.

Could you check it out and paste your already written documentation within the rust code so that we can have only one source of documentation.

**Comment by starlitxiling** [comment]
Okay, it looks great.

**Pull request merged**